### PR TITLE
Add PVnet weighted loss

### LINKED ISSUE #153: Add weighting quantile loss in training

### PR DESCRIPTION
**Pull request by jacobbieker**
# Pull Request

## Description

Adds weighted loss for quantile loss to help with longer forecast horizons, and ensuring the model focuses on shorter timescales as well.

Fixes #153 

## How Has This Been Tested?

Unit tests

- [x] Yes

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/.github/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: pvnet/models/base_model.py [hunk 1]
```diff
@@ -350,6 +353,8 @@ def _calculate_qauntile_loss(self, y_quantiles, y):
             errors = y - y_quantiles[..., i]
             losses.append(torch.max((q - 1) * errors, q * errors).unsqueeze(-1))
         losses = 2 * torch.cat(losses, dim=2)
+        if self.use_weighted_loss:
+            losses = losses * self.weighted_losses.weights
```
Comment (peterdudfield) [inline pvnet/models/base_model.py:357]: what are these weights?

Comment (jacobbieker) [reply pvnet/models/base_model.py:357]: These are from the WeightedLoss class, its from here: https://github.com/openclimatefix/PVNet/blob/f1534d8c01da5eb0bf8c0b001bc36aa05ced5c00/pvnet/models/utils.py#L129
